mongo/src/main/java/com/redhat/lightblue/mongo/crud/IterateAndUpdate.java

```diff
@@ -123,6 +124,8 @@ public void update(CRUDOperationContext ctx,
             measure.begin("iteration");
             int batchStartIndex=0; // docUpdateAttempts[batchStartIndex] is the first doc in this batch
             while (cursor.hasNext()) {
+                // Read from primary for read-for-update operations
+                cursor.setReadPreference(ReadPreference.primary());
```

Comment:
I think this needs to be set before the call to hasNext -- at that point we have already executed the query right?